pkg/osvscanner/osvscanner.go

```diff
@@ -608,7 +609,7 @@ func scanDebianDocker(r reporter.Reporter, dockerImageName string) ([]scannedPac
 }
 
 // Filters results according to config, preserving order. Returns total number of vulnerabilities removed.
-func filterResults(r reporter.Reporter, results *models.VulnerabilityResults, configManager *config.ConfigManager, allPackages bool) int {
+func filterResults(r reporter.Reporter, results *models.VulnerabilityResults, configManager *config.ConfigManager, allPackages bool, scanLicensesSummary bool) int {
```

Comment:
just coding through the consequences of this, we'll still need it in buildVulnerabilityResults as we need to be able to distinguish between --all-packages and --license-summary. If we just have --all-packages we don't want to include license info.

Reply:
we'll also still need it in DoScan to determine whether to make calls to deps.dev for licenses.

Reply:
but yeah, we won't need it in filterResults.

Reply:
After looking at what this would look like, I would prefer not to override --all-packages. I think it only saves some code just in filterResults at the cost of making it more confusing to understand - it's hard to remember that we overrode a flag all the way back in main.go. Thoughts?

Reply:
I think we should be overriding all-packages, since that's what is actually happening when we enable --license-summary right? (e.g. if we don't enable all-packages, and have just license summary, the JSON output will still have all packages listed correct?). 

I see it's only saving a couple of words in filterResults, but logically I think it makes it clearer what's happening.

Reply:
> if we don't enable all-packages, and have just license summary, the JSON output will still have all packages listed correct?

That's right.

I'll keep the flag overridden then.